llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -943,11 +955,24 @@ bool LoopVectorizationLegality::canVectorizeInstrs() {
       if (CI && !VFDatabase::getMappings(*CI).empty())
         VecCallVariantsFound = true;
 
+      auto canWidenInstruction = [this](Instruction const &Inst) {
+        Type *InstTy = Inst.getType();
+        if (isa<CallInst>(Inst) && isa<StructType>(InstTy) &&
+            canWidenCallReturnType(InstTy)) {
+          StructVecCallFound = true;
```

Comment:
Do we need this extra global state? Can we instead emit the new message for calls here when we encounter them, similar to other return types we cannot handle?

Reply:
Same question as https://github.com/llvm/llvm-project/pull/119221#discussion_r1890388807?

Reply:
Yep, lI thought I already submitted the comment, but couldn't find it in the UI